exercises/run-length-encoding/test/Tests.hs

```diff
@@ -0,0 +1,75 @@
+{-# LANGUAGE RecordWildCards #-}
+
+import Data.Foldable     (for_)
+import Test.Hspec        (Spec, describe, it, shouldBe)
+import Test.Hspec.Runner (configFastFail, defaultConfig, hspecWith)
+
+import RunLength (encode, decode)
+
+main :: IO ()
+main = hspecWith defaultConfig {configFastFail = True} specs
+
+specs :: Spec
+specs = describe "run-length-encoding" $ do
+          describe "decode" $ for_ decodeCases $ test decode
+          describe "encode" $ for_ encodeCases $ test encode
+          describe "both"   $ for_ bothCases   $ test (decode . encode)
+  where
+    test f Case{..} = it description $ f input `shouldBe` expected
+
+-- Test cases adapted from file
+-- `exercism/x-common/exercises/run-length-encoding/canonical-data.json`
+-- on 2016-12-26.
```

Comment:
I made a conscious decision to change the order of the tests. I like having tests for different methods separated so that I can fully implement `decode` and test it before I work on `encode`. I prefer this to testing that both `encode` and `decode` work for the simplest case, and then another simple case, etc. I feel this workflow is nicer for beginners, especially since we stop the test on first failure by default. On the other hand, it's a pain to manually reorder things. I assume that in the long run the exercism project will implement an automated system for generating the tests from the common json files though that's another story.